src/Lykke.Job.CandlesHistoryWriter.Services/Candles/CandlesSubscriber.cs

```diff
@@ -68,6 +75,9 @@ private async Task ProcessCandlesUpdatedEventAsync(CandlesUpdatedEvent candlesUp
         {
             try
             {
+                if (_cacheInitalizationService.InitializationState != CacheInitializationState.Idle)
+                    return;
```

Comment:
we should throw here to retry the message later, otherwise we will lost all messages recieved during cache initialization. To prevent a lot of retries spam, consider to add some delay before throwing the exception